Update OTel service entry span mapping docs (#23088)
DataDog/documentation: content/en/opentelemetry/schema_semantics/service_entry_spans.md

Datadog uses [service-entry spans][1] throughout the platform for features such as [Trace Metrics][2] and the [APM Trace Explorer][3]. This convention is unique to Datadog, but can be mapped from the [`SpanKind`][4] attribute in OpenTelemetry by following the opt-in guide below.

## Opting in to the feature

> This feature has been released with OTel Collector Contrib v0.100.0 and Datadog Agent v7.53.0. To opt into the **public beta**, enable the config option depending on the ingestion path.

Reviewer: ```suggestion
This feature requires OTel Collector Contrib v0.100.0 and Datadog Agent v7.53.0. To opt into the **public beta**, enable the config option depending on the ingestion path.
```

Author: I wasn't sure if I should say "requires" because this is the minimum version needed. Would you have any suggestions for this?

Reviewer: I've updated to add "or greater". This is how we usually document minimum required versions.
